File: app/app/src/main/java/com/example/hakonsreader/api/model/RedditVideo.java
package com.example.hakonsreader.api.model;

import com.google.gson.annotations.SerializedName;

public class RedditVideo {
    @SerializedName("duration")
    private int duration;

    @SerializedName("fallback_url")
    private String fallbackURL;

    @SerializedName("dash_url")
    private String dashURL;

    @SerializedName("hls_url")
    private String hlsURL;

    @SerializedName("height")
    private int height;

    @SerializedName("width")
    private int width;

    @SerializedName("is_gif")
    private boolean isGif;


    public int getDuration() {
        return duration;
    }

    /**
     * Gets the fallback URL for the video
     */
    public String getFallbackURL() {
        return fallbackURL;
    }

    /**
     * Gets the url to the DASH (Dynamic Adaptive Streaming over HTTP) video for the post
     */
    public String getDashUrl() {
        return dashURL;
    }

    /**
     * Gets the URL to the HLS (HTTP Live Streaming) video for the post
     */
    public String getHlsURL() {
        return hlsURL;
    }


    /**
     * Gets the width of the video
     */
    public int getWidth() {
        return width;
    }

    /**
     * Gets the height of the video
     */
    public int getHeight() {
        return height;
    }

    /**
     * @return True if the video is a gif
     */
    public boolean isGif() {
        return isGif;
    }
}

File: app/app/src/main/java/com/example/hakonsreader/api/model/RedditVideo.kt
package com.example.hakonsreader.api.model

import com.google.gson.annotations.SerializedName

class RedditVideo {
    @SerializedName("duration")
    val duration = 0

    /**
     * Gets the fallback URL for the video
     */
    @SerializedName("fallback_url")
    val fallbackUrl: String? = null

    /**
     * Gets the url to the DASH (Dynamic Adaptive Streaming over HTTP) video for the post
     */
    @SerializedName("dash_url")
    val dashUrl: String? = null

    /**
     * Gets the URL to the HLS (HTTP Live Streaming) video for the post
     */
    @SerializedName("hls_url")
    val hlsUrl: String? = null

    /**
     * Gets the height of the video
     */
    @SerializedName("height")
    val height = 0

    /**
     * Gets the width of the video
     */
    @SerializedName("width")
    val width = 0

    /**
     * @return True if the video is a gif
     */
    @SerializedName("is_gif")
    val isGif = false

    /**
     * The bitrate of the video in kilobits per second (kbps)
     */
    @SerializedName("bitrate_kbps")
    val bitrate = 0
}